ovh/resource_iam_policy.go

```diff
@@ -0,0 +1,170 @@
+package ovh
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceIamPolicy() *schema.Resource {
+	return &schema.Resource{
+		Importer: &schema.ResourceImporter{
+			State: func(rd *schema.ResourceData, i interface{}) ([]*schema.ResourceData, error) {
+				return []*schema.ResourceData{rd}, nil
+			},
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"identities": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"resources": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"allow": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"except": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"read_only": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+		ReadContext:   resourceIamPolicyRead,
+		CreateContext: resourceIamPolicyCreate,
+		UpdateContext: resourceIamPolicyUpdate,
+		DeleteContext: resourceIamPolicyDelete,
+	}
+}
+
+func resourceIamPolicyRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	config := meta.(*Config)
+
+	var pol IamPolicy
+	err := config.OVHClient.GetWithContext(ctx, fmt.Sprintf("/v2/iam/policy/%s", d.Id()), &pol)
```

Comment:
I often like to prevent path injection using `url.PathEscape`. And also, let's skip for `Sprintf ` as the formatting is just concat

```suggestion
	err := config.OVHClient.GetWithContext(ctx, "/v2/iam/policy/" + url.PathEscape(d.Id())), &pol)
```